### ISSUE TITLE
Make Standalone parser and extended grammar working together in some cases.

### ISSUE THREAD

**Issue by shiyunon** [lark-parser/lark#1060]
**Suggestion**

Currently, to extend a `statement` rule in a grammar called `old_gram.lark`, we can do `%import .old_gram.statement -> old_gram__statement` in the new grammar file `new_gram.lark`. The problem is that this does not work for Tokens and we cannot do `%import .old_gram.IDENTIFIER -> old_gram__IDENTIFIER` (lark will report an error because lark thinks `old_gram__IDENTIFIER` as a rule not a token). One workaround would be simply `%import .old_gram.IDENTIFIER` with the downside that ithe merged transformer of the old and the new grammar transformer would not work by default. 

There is also another big downside of the workground: if in `old_gram.lark`, we have the following rule:
```
identifier_list: IDENTIFIER ("," IDENTIFIER)*
```
and we import it to `new_gram.lark` with `%import .old_gram.identifier_list -> old_gram__identifier_list`. In the generated tree of programs, `IDENTIFIER` tokens will be translated to `old_gram__IDENTIFIER`.  We have no control of these `IDENTIFIER`s.

My suggestion is to either:
(1) support `%import .old_gram.IDENTIFIER -> old_gram__IDENTIFIER` explicitly.
(2) Make the default prefix of Token importing to be uppercase `%import .old_gram.IDENTIFIER -> OLD_GRAM__IDENTIFIER`

It would be great if someone can let me know if the above makes sense, I can probably help here too.

**Comment by erezsh** [comment]
There is the idea of making terminals MixedCase instead of only UPPER_CASE. That way, user can name them `MyTerminal` if they want, and it will also solve what you're referring to.

But I'm not sure how I feel about users importing rules and terminals into a namespace explicitly.

Why not have matching import from the transformer instead?

i.e.

```lark
%import .old_gram.IDENTIFIER 
```
and
```python
class NewTransformer(Transformer):
    from old_transformer import IDENTIFIER
```

**Comment by shiyunon** [comment]
Thanks for the reply. Yes, I agree that we can match import from the transformer, but have to do that is already a different user experience from importing rules. Also, this might worth being documented somewhere.

**Comment by erezsh** [comment]
Maybe so. This is a pretty new feature. I didn't even get a chance to use it much yet, so I can't say for sure what's the recommended usage.

**Comment by shiyunon** [comment]
Sorry, might have to confirm that I understood this:

> But I'm not sure how I feel about users importing rules and terminals into a namespace explicitly.

Importing rules into a namespace is required for many use cases, right? Based on https://github.com/lark-parser/lark/blob/master/examples/composition/storage.lark and https://github.com/lark-parser/lark/pull/973#issuecomment-907287565

**Comment by erezsh** [comment]
Yes, importing rules in the local namespace is a common operation.

What you were doing is injecting rules into other namespaces. In this case, the original namespace, so it's not *as* bad, but still feels like it shouldn't be encouraged. I'd much rather add a `%soft_import` directive that doesn't remove the namespace.
